modal_server_vllm.py

```diff
@@ -0,0 +1,424 @@
+from http import HTTPStatus
+import os
+import time
+import uuid
+from typing import Annotated, Dict, List, Literal, Optional, Union
+
+from fastapi.responses import JSONResponse
+import modal
+from fastapi import FastAPI, Header, Request
+from pydantic import BaseModel, Field, field_validator, validator
+
+from functionary.openai_types import (
+    ChatMessage,
+    Function,
+    Tool,
+)
+
+
+
+app = modal.App("functionary_vllm")
+fast_api_app = FastAPI(title="Functionary API")
+
+#MODEL = "meetkai/functionary-small-v2.4"
+MODEL = "meetkai/functionary-medium-v2.4"
+MAX_MODEL_LENGTH = 8196
+LOADIN8BIT = False
+#GPU_CONFIG = modal.gpu.L4(count=1)
+GPU_CONFIG = modal.gpu.A100(memory=80, count=2)
```

Comment:
Can we use a config file or command-line arguments for these variables instead of manually changing the script itself? For the GPU_CONFIG, can we have a dictionary that maps from model name to the GPU required? The correct GPU resources will be used when the appropriate model name is provided.

Reply:
Added modal_server_config.py for modal server configuration